### ISSUE TITLE
Serializing ensemble models

### ISSUE THREAD

**Issue by eduardowoj** [creme-ml/creme#164]
Hello,

I am developing a project using the RandomForestClassifier newly introduced in creme 0.4, but I'm stuck on a problem.  I can't serialize the trained models for testing and deploying, neither with the `pickle` module or the most powerful `dill` (examples given below). I believe this is caused by the implementation of the ensemble wrapper, so when the interpreter tries to pickle the model, it only finds the wrapper and not the individual models.

This happens just with the ensemble models. The base models like logistic regression and decision trees can be pickled without any problem.

I am running creme 0.4.2 under conda 4.7.12, with dill 0.3.0 and Python 3.6.9 (anaconda distribution), under Ubuntu Linux 18.04 (LTS) x64. 

```
pickle.dump(model,open('model1.pkl','wb'))
---------------------------------------------------------------------------
PicklingError                             Traceback (most recent call last)
<ipython-input-108-3df7192e8e70> in <module>
----> 1 pickle.dump(model,open('model1.pkl','wb'))

PicklingError: args[0] from __newobj__ args has the wrong class
```

```
dill.dump(model,open('model1.pkl','wb'))
---------------------------------------------------------------------------
PicklingError                             Traceback (most recent call last)
<ipython-input-109-de2aaa6a3d66> in <module>
----> 1 dill.dump(model,open('model1.pkl','wb'))

~/anaconda3/lib/python3.6/site-packages/dill/_dill.py in dump(obj, file, protocol, byref, fmode, recurse)
    285         raise PicklingError(msg)
    286     else:
--> 287         pik.dump(obj)
    288     stack.clear()  # clear record of 'recursion-sensitive' pickled objects
    289     return

~/anaconda3/lib/python3.6/pickle.py in dump(self, obj)
    407         if self.proto >= 4:
    408             self.framer.start_framing()
--> 409         self.save(obj)
    410         self.write(STOP)
    411         self.framer.end_framing()

~/anaconda3/lib/python3.6/pickle.py in save(self, obj, save_persistent_id)
    519 
    520         # Save the reduce() output and finally memoize the object
--> 521         self.save_reduce(obj=obj, *rv)
    522 
    523     def persistent_id(self, obj):

~/anaconda3/lib/python3.6/pickle.py in save_reduce(self, func, args, state, listitems, dictitems, obj)
    601             if obj is not None and cls is not obj.__class__:
    602                 raise PicklingError(
--> 603                     "args[0] from __newobj__ args has the wrong class")
    604             args = args[1:]
    605             save(cls)

PicklingError: args[0] from __newobj__ args has the wrong class
```

**Comment by MaxHalford** [comment]
Thanks for opening the issue. In `creme` we use a lot of inheritance, and this has consequences on the pickling mechanism. I'll take a look at this ASAP. Also this should be caught by unit tests, but obviously hasn't.

**Comment by MaxHalford** [comment]
Fixed in 4aa28e8.
